lib/authorize_net/api.rb

```diff
@@ -280,21 +280,17 @@ def handleResponse(raw_response)
   #
   # @throws AuthorizeNet::Exception
   # =============================================
-  def validate_hash(response_xml, amount, options={})
-    if @md5_hash.nil?
+  def validate_hash(response_xml, amount)
+    if @signature_key.nil?
```

Comment:
This seems fine?  Although I can also imagine taking this out and enforcing hash validation